#159236204-Allow comparison of training members

solnsubuga opened the pull request:
#### What does this PR do?
- Allows comparison of two users who might be training together based on weight entry data.
#### Description of Task to be completed?
- Allow a trainer to select two or more users and compare them.
#### How should this be manually tested?
-  Checkout the branch `ft-compare-members-159236204`
- Run the command `sudo python manage.py bower_install --allow-root `
- Run `cd ./extras/dummy_generator`
- To add some dummy members  run `python generator.py users 10`
- Add some weights `python generator.py weight 10`
- Run the application `python manage.py runserver`
- Navigate to gym members page
- Select two or more members from the members' list
- Click the compare members button
#### What are the relevant pivotal tracker stories?
- [#159236204](https://www.pivotaltracker.com/story/show/159236204)
#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-08-22 at 14 24 03" src="https://user-images.githubusercontent.com/23418080/44460791-23ce9580-a617-11e8-929b-815470b9cc60.png">
<img width="1440" alt="screen shot 2018-08-22 at 16 04 43" src="https://user-images.githubusercontent.com/23418080/44465079-2df79080-a625-11e8-9d42-56ce555e1550.png">

wasswa-derick commented on wger/weight/views.py:
```diff
@@ -190,6 +191,29 @@ def get_weight_data(request, username=None):
 
+@api_view(['GET'])
+def get_user_weight_data(request):
+    q = request.GET.get('usernames', '')
```
Variable name q could be made more descriptive.

solnsubuga replied:
I have changed it to usernames_param.

wasswa-derick commented on wger/gym/templates/gym/member_list.html:
```diff
@@ -139,7 +139,108 @@ <h5 class="modal-title" id="delete-modal-title" style="font-size: 140%; color: #
+        var colors = [
+        "#FFB81D","#AA0050","#0033a1","#665EC7","#7C868D","#00a1e0","#0a2240",
+        "#FF681D","#00A1B0","#B150C5",
+
```
Blank line not useful here.

solnsubuga replied:
Removed it.

wasswa-derick reviewed (approved):
LGTM

wasswa-derick commented on wger/gym/templates/gym/member_list.html:
```diff
@@ -139,7 +139,107 @@ <h5 class="modal-title" id="delete-modal-title" style="font-size: 140%; color: #
+    function selectUser(event,id){
+        if(event.checked == true){
+            selectedUsers.push(id);
+        }else{
```
Add spaces between the operators and the keyword 'else'.

solnsubuga committed:
feat(comparison) Allow comparison of members

Add endpoint for getting weight data for members to compare
Add functionality to generate graph.

[Delivers #159236204]
